pkg/rpcclient/wsclient.go

```diff
@@ -539,6 +539,12 @@ readloop:
 	}
 	c.respChannels = nil
 	c.respLock.Unlock()
+	c.subscriptionsLock.Lock()
+	for rcvr, ids := range c.receivers {
+		c.subscriptions[ids[0]].Close()
+		delete(c.receivers, rcvr)
+	}
+	c.subscriptionsLock.Unlock()
```

Comment:
Initially the meaning of receiver channel closed by WSClient was "missed event is received, thus I'll close the channel to notify the receiver". Here we do change the meaning of closing the receiver's channel by WSClient. Now it means that "WSClient stops his routines". IMHO, we have to change the documentation, currently we have:
```
// The receiver channel will be closed by the WSClient
// immediately after MissedEvent is received from the server; no unsubscription
// is performed in this case, so it's the user responsibility to unsubscribe.
```
And secondly, we should take care of naive receiver. Current implementation of `Close()` for naive receiver sends `MissedEvent` to c.Notifications, see the https://github.com/nspcc-dev/neo-go/blob/7bc9b6e3236e4e89aea3ef86240746f3d17e591f/pkg/rpcclient/wsclient.go#L329 This behaviour is backwards-compatible and should be kept until `c.Notifications` removal. However, I don't think that sending missed event in the current context is the right thing to do. We probably should filter out naive receivers here.

Reply:
> Now it means that "WSClient stops his routines"

It can be both ways. "Something is wrong with the subscription, go figure".

Reply:
> It can be both ways.

Yes, but the documentation says only about the "missed event" case.

Reply:
Updated.